Examples/DataSender.agent.nut

```diff
@@ -0,0 +1,98 @@
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+#require "MongoDBStitch.agent.lib.nut:1.0.0"
+
+// MongoDBStitch library sample.
+// Periodically sends data to MongoDB using pre-configured MongoDB Stitch named pipeline.
+// Data are sent every 10 seconds, they consist of integer increasing value, converted to string
+// and measureTime attribute with data measurement time in seconds since the epoch format.
+
+const MONGO_DB_STITCH_INSERT_DATA_PIPELINE = "testInsertData";
+const SEND_DATA_PERIOD = 10.0;
+
+class DataSender {
+    _counter = 0;
+    _stitchClient = null;
+    _apiKey = null;
+
+    constructor(appId, apiKey) {
+        _stitchClient = MongoDBStitch(appId);
+        _apiKey = apiKey;
+    }
+
+    // Returns a data to be sent
+    function getData() {
+        _counter++;
+        return {
+            "data" : {
+                "value" : _counter.tostring(),
+                "measureTime" : time().tostring()
+            }
+        };
+    }
+
+    // Periodically sends data to MongoDB using pre-configured MongoDB Stitch named pipeline
+    function sendData() {
+        local data = getData();
+        _stitchClient.executeNamedPipeline(
+            MONGO_DB_STITCH_INSERT_DATA_PIPELINE,
+            data,
+            function (error, response) {
+                if (error) {
+                    server.error("Data insertion failed: " + error.details);
+                } else {
+                    server.log("Data inserted successfully:");
+                    server.log(http.jsonencode(data.data));
+                }
+            }.bindenv(this));
+
+        imp.wakeup(SEND_DATA_PERIOD, function () {
+            sendData();
+        }.bindenv(this));
+    }
+
+    // Logins to the Stitch application and starts data sending
+    function start() {
+        _stitchClient.loginWithApiKey(_apiKey, function (error, response) {
+            if (error) {
+                server.error("MongoDB Stitch authentication failed: " + error.details);
+            } else {
+                sendData();
+            }
+        }.bindenv(this));
+    }
+}
+
+// RUNTIME
+// ---------------------------------------------------------------------------------
+
+// MONGO DB STITCH CONSTANTS
+// ----------------------------------------------------------
+const MONGO_DB_STITCH_APPLICATION_ID = "<YOUR_STITCH_APPLICATION_ID>";
+const MONGO_DB_STITCH_API_KEY = "<YOUR_STITCH_API_KEY>";
```

Comment:
[q] Is this kind of practice okay for electricimp? I'm not sure if people would accidentally commit this to github or not or if there is a way to have secure data configuration.

Reply:
Hi @edaniels, yes this is a common practice (not ideal but this is what it is). In fact you can use https://github.com/electricimp/builder not to hardcode things in the application code and parameterize it at "build" time (which is the preferred/recommended but more sophisticated way).

Reply:
@edaniels fixed.